bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/cli/CliUtil.java

```diff
@@ -66,12 +67,24 @@ static Object getBValue(Type type, String value, String parameterName) {
         switch (type.getTag()) {
             case TypeTags.STRING_TAG:
                 return StringUtils.fromString(value);
+            case TypeTags.CHAR_STRING_TAG:
+                return getCharValue(value, parameterName);
             case TypeTags.INT_TAG:
+            case TypeTags.SIGNED32_INT_TAG:
+            case TypeTags.SIGNED16_INT_TAG:
+            case TypeTags.SIGNED8_INT_TAG:
+            case TypeTags.UNSIGNED32_INT_TAG:
+            case TypeTags.UNSIGNED16_INT_TAG:
+            case TypeTags.UNSIGNED8_INT_TAG:
                 return getIntegerValue(value, parameterName);
+            case TypeTags.BYTE_TAG:
+                return getByteValue(value, parameterName);
             case TypeTags.FLOAT_TAG:
                 return getFloatValue(value, parameterName);
             case TypeTags.DECIMAL_TAG:
                 return getDecimalValue(value, parameterName);
+            case TypeTags.TYPE_REFERENCED_TYPE_TAG:
+                return getBValue(TypeUtils.getReferredType(type), value, parameterName);
```

Comment:
Shall we add a test to cover this and above type reference case?

Reply:
This line is executed through the tests. If I remove this case, several built-in subtype tests are failing.